src/xtgeo/grid3d/_grid_wellzone.py

```diff
@@ -1,29 +1,39 @@
 """Private module for grid vs well zonelog checks."""
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Literal
 
 import numpy as np
 
-import xtgeo
 from xtgeo.common import XTGeoDialog, null_logger
+from xtgeo.grid3d import GridProperty
+from xtgeo.well import Well
+
+if TYPE_CHECKING:
+    from xtgeo.grid3d import Grid
+
 
 xtg = XTGeoDialog()
 logger = null_logger(__name__)
 
 
 def report_zone_mismatch(
-    self,
-    well=None,
-    zonelogname="ZONELOG",
-    zoneprop=None,
-    onelayergrid=None,  # redundant; will be computed internally
-    zonelogrange=(0, 9999),
-    zonelogshift=0,
-    depthrange=None,
-    perflogname=None,
-    perflogrange=(1, 9999),
-    filterlogname=None,
-    filterlogrange=(1e-32, 9999.0),
-    resultformat=1,
-):
+    self: Grid,
+    well: Well | None = None,
+    zonelogname: str = "ZONELOG",
+    zoneprop: GridProperty | None = None,
+    onelayergrid: Grid | None = None,  # redundant; will be computed internally
+    zonelogrange: tuple[int, int] = (0, 9999),
+    zonelogshift: int = 0,
+    depthrange: tuple[int | float, int | float] | None = None,
+    perflogname: str | None = None,
+    perflogrange: tuple[int | float, int | float] = (1, 9999),
+    filterlogname: str | None = None,
+    filterlogrange: tuple[int | float, int | float] = (1e-32, 9999.0),
+    resultformat: Literal[1, 2] = 1,
```

Comment:
Not sure if this is correct.. the documentation says that the two options are 1 or 2, but the implementation will only do a specific thing for `resultformat=1`